core/src/database/DatabaseSessionPool.cpp

```diff
@@ -52,22 +52,33 @@ namespace ledger {
                 explicit SessionTracer(std::shared_ptr<api::CoreTracer> tracer) : _tracer(std::move(tracer)) {
                 }
 
-                void startSpan(std::string &string) override {
-                    _span = _tracer->startSpan(string);
-                    _span->setTagStr("db.statement", string);
-                    _span->setTagStr("sql.query", string);
-                    _span->setTagStr("db.system", "postgresql");
-                    _span->setTagStr("span.type", "sql");
-                    _span->setTagStr("service", "postgresql");
+                int startSpan(std::string &string) override {
+                    const std::lock_guard<std::mutex> lock(mut);
+                    auto span = _tracer->startSpan(string);
+                    span->setTagStr("db.statement", string);
+                    span->setTagStr("sql.query", string);
+                    span->setTagStr("db.system", "postgresql");
+                    span->setTagStr("span.type", "sql");
+                    span->setTagStr("service", "postgresql");
+                    int id = idx++;
```

Comment:
`const std::lock_guard<std::mutex> lock(mut);` can be moved right before this line